### ISSUE TITLE
Unable to resolve module `AccessibilityInfo` on 0.57.2 iOS

### ISSUE THREAD

**Issue by jinxac** [facebook/react-native#21752]
## Environment

```
  React Native Environment Info:
    System:
      OS: macOS Sierra 10.12.6
      CPU: x64 Intel(R) Core(TM) i5 CPU       M 540  @ 2.53GHz
      Memory: 26.35 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 9.9.0 - /usr/local/bin/node
      Yarn: 1.3.2 - /usr/local/bin/yarn
      npm: 3.10.10 - /usr/local/bin/npm
      Watchman: 4.7.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 11.2, macOS 10.13, tvOS 11.2, watchOS 4.2
      Android SDK:
        Build Tools: 23.0.1, 23.0.2, 23.0.3, 24.0.0, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.0, 26.0.1, 26.0.2, 26.0.3, 27.0.0, 27.0.1, 27.0.2, 27.0.3
        API Levels: 23, 24, 25, 26, 27
    IDEs:
      Xcode: 9.2/9C40b - /usr/bin/xcodebuild
    npmPackages:
      react: 16.5.0 => 16.5.0
      react-native: 0.57.2 => 0.57.2
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-create-library: 3.1.2
      react-native-git-upgrade: 0.2.7
      react-native-rename: 2.2.2
```

## Description

Upgraded from 0.48.3 to 0.57.2 and on iOS building code throws error `unable to resolve module AccessibilityInfo`. Android works fine. This issue was resolved in 0.56.1 according to the bugs I saw in the issue list but I am still facing this issue hence opened a new one.

```
error: bundling failed: Error: Unable to resolve module `AccessibilityInfo` from `/Users/harkirat/wealthy/bumblebee-new/node_modules/react-native/Libraries/react-native/react-native-implementation.js`: Module `AccessibilityInfo` does not exist in the Haste module map

This might be related to https://github.com/facebook/react-native/issues/4968
To resolve try the following:
  1. Clear watchman watches: `watchman watch-del-all`.
  2. Delete the `node_modules` folder: `rm -rf node_modules && npm install`.
  3. Reset Metro Bundler cache: `rm -rf /tmp/metro-bundler-cache-*` or `npm start -- --reset-cache`.
  4. Remove haste cache: `rm -rf /tmp/haste-map-react-native-packager-*`.
    at ModuleResolver.resolveDependency (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:209:1301)
    at ResolutionRequest.resolveDependency (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/node-haste/DependencyGraph/ResolutionRequest.js:83:16)
    at DependencyGraph.resolveDependency (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/node-haste/DependencyGraph.js:222:485)
    at Object.resolve (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/lib/transformHelpers.js:149:25)
    at dependencies.map.result (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/DeltaBundler/traverseDependencies.js:316:29)
    at Array.map (<anonymous>)
    at resolveDependencies (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/DeltaBundler/traverseDependencies.js:312:16)
    at /Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/DeltaBundler/traverseDependencies.js:169:33
    at Generator.next (<anonymous>)
    at step (/Users/harkirat/wealthy/bumblebee-new/node_modules/metro/src/DeltaBundler/traverseDependencies.js:271:307)
```

Following is the list of devDependencies in package.json

```
  "devDependencies": {
    "@babel/cli": "^7.0.0",
    "@babel/core": "^7.0.0",
    "@babel/plugin-proposal-decorators": "^7.1.2",
    "babel-core": "^7.0.0-bridge.0",
    "babel-eslint": "7.1.1",
    "babel-jest": "18.0.0",
    "babel-loader": "^8.0.4",
    "babel-plugin-transform-remove-console": "6.8.5",
    "babel-preset-react-native": "4.0.0",
    "eslint": "3.13.1",
    "eslint-config-eslint": "3.0.0",
    "eslint-friendly-formatter": "2.0.7",
    "eslint-loader": "1.6.1",
    "eslint-plugin-import": "2.2.0",
    "eslint-plugin-jsx-a11y": "3.0.2",
    "eslint-plugin-react": "7.5.1",
    "jest": "18.0.0",
    "metro-react-native-babel-preset": "^0.48.1",
    "react-native-schemes-manager": "1.0.1",
    "react-test-renderer": "15.4.1"
  },
```

**Comment by jaycanty** [comment]
So for me....

```

**Comment by jinxac** [comment]
@jaycanty android is working fine for me, so i am not really sure downgrading is an option, but yeah spent almost a day trying to figure it out.

**Comment by jaycanty** [comment]
@jinxac I can run in both iOS and Android, I just can't run tests...

**Comment by pacocd** [comment]
Same here, I'm trying to run tests in React Native 0.57.2, I replaced ```module:metro-react-native``` with ```react-native``` in ```.babelrc``` and got the same above errors.

**Comment by pacocd** [comment]
@motogod the issue is not happening in 0.55.4. Just downgrade RN version and replace ```module:metro-react-native``` with ```react-native``` in your ```.babelrc```.

That worked for me.

**Comment by jinxac** [comment]
After trying pretty much everything on javascript side I  did following :-  

1. Created a new react native project
2. Copied the js folder here
3. Linked all the packages

Everything was working fine here with the new project. The only difference in my old project was in `AppDelegate.m`.

I have integrated `code-push` to my project. In my old project I had following code for `jsCodeLocation` (**without query parameters**). It was working just fine in **0.48.3**

```
  NSURL *jsCodeLocation;
    #ifdef DEBUG
      jsCodeLocation = [NSURL URLWithString:@"http://192.168.1.39:8081/index.bundle];
    #else
      jsCodeLocation = [CodePush bundleURL];
    #endif
```

Now when I migrated to **0.57.2** I blindly copied the same thing. It started throwing error because of this line :- 

```
 jsCodeLocation = [NSURL URLWithString:@"http://192.168.1.39:8081/index.bundle];  // throws error
 jsCodeLocation = [NSURL URLWithString:@"http://192.168.1.39:8081/index.bundle/? 
 platform=ios&dev=true];  //works perfectly fine
```

It was working fine on new project because `react-native` creates the project with following code:-

```
jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
```

Once I added remote ip to debug remotely it started throwing the error mentioned in the title. Phew!!

I would not have been to resolve this unless I had seen [https://github.com/facebook/react-native/issues/21303](https://github.com/facebook/react-native/issues/21303). It convinced me that might be something wrong on native iOS side only as android is working fine(javascript issue should have broken both). Also there were no issues were there in build logs. 


**P.S**:-

1. I am not sure how it was working earlier in **0.48.3** and what changes made this to break in current version.  I could not find this anywhere in the **documentation**. 
2. Also, the error messages should be a bit more **clear**, this was like finding a needle in a haystack and it took almost 5 days of effort to find this :( . 
2. Much thanks @hramos for describing the issue in detail.

I really hope it really helps someone.

**Comment by victorgutemberg** [comment]
Just tried running the steps described [here](https://github.com/facebook/react-native/issues/21752#issuecomment-429433589 ) and the issue still happens.
Actually, while I was searching for a solution I found [another issue](https://github.com/facebook/react-native/issues/14209) describing the same situation. This issue seems to have been around since May 2017.

I'm developing only for android, but the issue happens when running tests.

 ```
 React Native Environment Info:
    System:
      OS: macOS Sierra 10.12.6
      CPU: x64 Intel(R) Core(TM) i5-4278U CPU @ 2.60GHz
      Memory: 32.27 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.11.0 - /usr/local/bin/node
      Yarn: 1.9.4 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 10.3, macOS 10.12, tvOS 10.2, watchOS 3.2
      Android SDK:
        Build Tools: 19.0.0, 19.1.0, 21.1.2, 22.0.1, 23.0.1, 26.0.2, 27.0.2, 27.0.3, 28.0.2
        API Levels: 19, 21, 22, 23, 26, 27, 28
    IDEs:
      Android Studio: 3.1 AI-173.4819257
      Xcode: 8.3.2/8E2002 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.0-alpha.8af6728 => 16.6.0-alpha.8af6728 
      react-native: 0.57.4 => 0.57.4 
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

**Comment by jaycanty** [comment]
I was able to get tests running by adding `transform` to `jest` in `package.json`....

```
  "jest": {
    "preset": "react-native",
    "transform": {
      "^.+\\.(js)$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
    }
  }
```

Sorry I can explain what is going on here but it worked....
